model/riscv_insts_cext.sail

```diff
@@ -426,201 +428,202 @@ function clause execute (C_ADDW(rsd, rs2)) = {
+      if nzui5 @ nzui40 != 0b000000 & rsd != zreg & (sizeof(xlen) == 64 | nzui5 == 0b0) & extension("C")
   <-> 0b000 @ nzui5 : bits(1) @ rsd : regidx @ nzui40 : bits(5) @ 0b10
-      if nzui5 @ nzui40 != 0b000000 & rsd != zreg & (sizeof(xlen) == 64 | nzui5 == 0b0)
+      if nzui5 @ nzui40 != 0b000000 & rsd != zreg & (sizeof(xlen) == 64 | nzui5 == 0b0) & extension("C")
 
 function clause execute (C_SLLI(shamt, rsd)) =
   execute(SHIFTIOP(shamt, rsd, rsd, RISCV_SLLI))
 
 mapping clause assembly = C_SLLI(shamt, rsd)
-      if shamt != 0b000000 & rsd != zreg
+      if shamt != 0b000000 & rsd != zreg & extension("C")
   <-> "c.slli" ^ spc() ^ reg_name(rsd) ^ sep() ^ hex_bits_6(shamt)
-      if shamt != 0b000000 & rsd != zreg
+      if shamt != 0b000000 & rsd != zreg & extension("C")
 
 /* ****************************************************************** */
 union clause ast = C_LWSP : (bits(6), regidx)
 
 mapping clause encdec_compressed = C_LWSP(ui76 @ ui5 @ ui42, rd)
-      if rd != zreg
+      if rd != zreg & extension("C")
   <-> 0b010 @ ui5 : bits(1) @ rd : regidx @ ui42 : bits(3) @ ui76 : bits(2) @ 0b10
-      if rd != zreg
+      if rd != zreg & extension("C")
 
 function clause execute (C_LWSP(uimm, rd)) = {
   let imm : bits(12) = zero_extend(uimm @ 0b00);
   execute(LOAD(imm, sp, rd, false, WORD, false, false))
 }
 
 mapping clause assembly = C_LWSP(uimm, rd)
-      if rd != zreg
+      if rd != zreg & extension("C")
   <-> "c.lwsp" ^ spc() ^ reg_name(rd) ^ sep() ^ hex_bits_6(uimm)
-      if rd != zreg
+      if rd != zreg & extension("C")
 
 /* ****************************************************************** */
 union clause ast = C_LDSP : (bits(6), regidx)
 
 mapping clause encdec_compressed = C_LDSP(ui86 @ ui5 @ ui43, rd)
-      if rd != zreg & sizeof(xlen) == 64
+      if rd != zreg & sizeof(xlen) == 64 & extension("C")
   <-> 0b011 @ ui5 : bits(1) @ rd : regidx @ ui43 : bits(2) @ ui86 : bits(3) @ 0b10
-      if rd != zreg & sizeof(xlen) == 64
+      if rd != zreg & sizeof(xlen) == 64 & extension("C")
 
 function clause execute (C_LDSP(uimm, rd)) = {
   let imm : bits(12) = zero_extend(uimm @ 0b000);
   execute(LOAD(imm, sp, rd, false, DOUBLE, false, false))
 }
 
 mapping clause assembly = C_LDSP(uimm, rd)
-      if rd != zreg & sizeof(xlen) == 64
+      if rd != zreg & sizeof(xlen) == 64 & extension("C")
   <-> "c.ldsp" ^ spc() ^ reg_name(rd) ^ sep() ^ hex_bits_6(uimm)
-      if rd != zreg & sizeof(xlen) == 64
+      if rd != zreg & sizeof(xlen) == 64 & extension("C")
 
 /* ****************************************************************** */
 union clause ast = C_SWSP : (bits(6), regidx)
 
-mapping clause encdec_compressed = C_SWSP(ui76 @ ui52, rs2)
-  <-> 0b110 @ ui52 : bits(4) @ ui76 : bits(2) @ rs2 : regidx @ 0b10
+mapping clause encdec_compressed = C_SWSP(ui76 @ ui52, rs2) if extension("C") if extension("C")
+  <-> 0b110 @ ui52 : bits(4) @ ui76 : bits(2) @ rs2 : regidx @ 0b10 if extension("C") if extension("C")
```

Comment:
You have double adds here.

Reply:
Yes I overlooked those lines, I've fixed it. Let me know if any further changes required